### PR TITLE
feat(types,clerk-js): Add support for backup code strategy in sign in flow

### PR DESCRIPTION
**Pull request by chanioxaris**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

- Introduce the necessary types to support the new backup_code strategy
- Support for the new strategy in sign in second factor flow
- Introduce the new sign in card for backup code

### Before

Not supported

### After

https://user-images.githubusercontent.com/22435234/191707282-9fca82e7-4600-4498-a153-53f379561f4c.mov

https://www.notion.so/clerkdev/Support-backup-code-strategy-in-clerk-js-sign-in-flow-and-components-28a72a5a572a437cac5546b1f78aa406

### CODE REVIEW

File: packages/clerk-js/src/ui/SignIn/SignInFactorTwoAlternativeMethods.tsx [hunk 1]
```diff
@@ -80,6 +80,8 @@ export function getButtonLabel(factor: SignInFactor): string {
       return `Send code to ${formatSafeIdentifier(factor.safeIdentifier)}`;
     case 'totp':
       return 'Use your authenticator app';
+    case 'backup_code':
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/SignIn/SignInFactorTwoAlternativeMethods.tsx:83]: 🛑 These should be returning localization keys, not strings. I'll handle this right away with a new commit.

File: packages/clerk-js/src/ui/SignIn/SignInFactorTwoBackupCodeCard.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  const { onShowAlternativeMethodsClicked } = props;
+  const signIn = useCoreSignIn();
+  const { displayConfig } = useEnvironment();
+  const { navigateAfterSignIn } = useSignInContext();
+  const { setActive } = useCoreClerk();
+  const supportEmail = useSupportEmail();
+  const card = useCardState();
+  const codeControl = useFormControl('code', '', {
+    type: 'text',
+    label: '',
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply packages/clerk-js/src/ui/SignIn/SignInFactorTwoBackupCodeCard.tsx]: Yes my bad, added fe44890c51cdee8cb2bdb3d4dc5adf35fe0b57e4

File: packages/clerk-js/src/ui/SignIn/SignInFactorTwoAlternativeMethods.tsx (hunk 1 above)
Comment (chanioxaris) [reply packages/clerk-js/src/ui/SignIn/SignInFactorTwoAlternativeMethods.tsx:83]: I can do it as well in a separate PR if you would like

Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/SignIn/SignInFactorTwoAlternativeMethods.tsx:87]: Posting here for extra visibility, I handled the missing keys with a separate commit. Thanks Hari!

**Review by nikosdouvlis — APPROVED** [review]

**Pull request merged**